test/unit/useQuery.test.js

```diff
@@ -1,5 +1,123 @@
-import { useQuery } from '../../src';
+import React from 'react';
+import { testHook } from 'react-testing-library';
+import { ClientContext, useQuery, useClientRequest } from '../../src';
+
+jest.mock('../../src/useClientRequest');
+
+let mockQueryReq, mockState, mockClient;
+
+const Wrapper = props => (
+  <ClientContext.Provider value={mockClient}>
+    {props.children}
+  </ClientContext.Provider>
+);
+
+const TEST_QUERY = `query Test($limit: Int) {
+  tests(limit: $limit) {
+    id
+  }
+}`;
 
 describe('useQuery', () => {
-  it('runs a test', () => {});
+  beforeEach(() => {
+    mockQueryReq = jest.fn();
+    mockState = { loading: true, cacheHit: false };
+    useClientRequest.mockReturnValue([mockQueryReq, mockState]);
+
+    mockClient = {
+      ssrMode: false,
+      ssrPromises: []
+    };
+  });
+
+  afterEach(() => {
+    jest.unmock('../../src/useClientRequest');
+  });
+
+  it('calls useClientRequest with query', () => {
+    testHook(() => useQuery(TEST_QUERY), { wrapper: Wrapper });
+    expect(useClientRequest).toHaveBeenCalledWith(TEST_QUERY, {
+      useCache: true
+    });
+  });
+
+  it('calls useClientRequest with options', () => {
+    testHook(() => useQuery(TEST_QUERY, { useCache: false, extra: 'option' }), {
+      wrapper: Wrapper
+    });
+    expect(useClientRequest).toHaveBeenCalledWith(TEST_QUERY, {
+      useCache: false,
+      extra: 'option'
+    });
+  });
+
+  it('returns initial state from useClientRequest & refetch', () => {
+    let state;
+    testHook(() => (state = useQuery(TEST_QUERY)), { wrapper: Wrapper });
+    expect(state).toEqual({
+      loading: true,
+      cacheHit: false,
+      refetch: expect.any(Function)
+    });
+  });
+
+  it('bypasses cache when refetch is called', () => {
+    let refetch;
+    testHook(() => ({ refetch } = useQuery(TEST_QUERY)), { wrapper: Wrapper });
+    refetch();
+    expect(mockQueryReq).toHaveBeenCalledWith({ skipCache: true });
+  });
+
+  it('sends the query on mount if no data & no error', () => {
+    testHook(() => useQuery(TEST_QUERY), { wrapper: Wrapper });
+    expect(mockQueryReq).toHaveBeenCalledTimes(1);
+  });
+
+  it('does not send the same query on mount if data is already present', () => {
+    mockState.data = { some: 'data' };
+    testHook(() => useQuery(TEST_QUERY), { wrapper: Wrapper });
+    expect(mockQueryReq).not.toHaveBeenCalled();
+  });
+
+  it('does not send the query on mount if there is already data', () => {
+    mockState.error = true;
+    testHook(() => useQuery(TEST_QUERY), { wrapper: Wrapper });
+    expect(mockQueryReq).not.toHaveBeenCalled();
+  });
+
+  it('does not send the query on mount if there is an error', () => {
```

Comment:
Ah this is the correct test - the one above is a duplicate.

Reply:
Nice spot, removed this